### ISSUE TITLE
geyser plugin triggers the server status to be online, but the server has not finished loading yet

### ISSUE THREAD

**Issue by debian-user-france1** [gekigek99/minecraft-server-hibernation#112]
I have geyser for bedrock compatibility. Geyser has an equal start message as minecraft so msh thinks the server is startet while other plugins are loading. While that happens players try to join and msh registers as joined. They get kicked because - as we know - server isn't startet. msh doesn't de-register them so if they join afterwards we have too many players joined that are actually not in:

```
[22:31:50] [Server thread/INFO]: [Geyser-Spigot] Auto-loaded Floodgate key
[22:31:50] [Server thread/INFO]: [Geyser-Spigot] ******************************************
[22:31:50] [Server thread/INFO]: [Geyser-Spigot] 
[22:31:50] [Server thread/WARN]: [Geyser-Spigot] de_CH is not a valid Bedrock language.
[22:31:50] [Server thread/INFO]: [Geyser-Spigot] Loading Geyser version 1.4.1-SNAPSHOT (git-master-070ff46)
[22:31:50] [Server thread/INFO]: [Geyser-Spigot] 
[22:31:50] [Server thread/INFO]: [Geyser-Spigot] ******************************************
[22:32:04] [Server thread/INFO]: [Geyser-Spigot] Loaded Floodgate key!
[22:32:05] [Network Listener - #1/INFO]: [Geyser-Spigot] Started Geyser on 0.0.0.0:19132
[22:32:05] [Server thread/INFO]: [Geyser-Spigot] Done (14.522s)! Run /geyser help for help!
2021/07/27 22:32:05 *** MINECRAFT SERVER IS ONLINE!
[22:32:05] [Server thread/INFO]: [HolographicDisplays] Enabling HolographicDisplays v2.4.8
[22:32:05] [Server thread/INFO]: [HolographicDisplays] Enabled player relative placeholders with ProtocolLib.
[22:32:06] [Server thread/INFO]: Server permissions file permissions.yml is empty, ignoring it
[22:32:06] [Server thread/INFO]: Done (74.035s)! For help, type "help"
```
Here you see: Geyser causes a startup trigger

**Comment by gekigek99** [comment]
is it normal to have a log like this? it is pretty confusing ahahah :)

**Comment by gekigek99** [comment]
`[22:32:06] [Server thread/INFO]: Done (74.035s)! For help, type "help"`

this is the line when the server is really online?

**Comment by gekigek99** [comment]
can you show a log example of when a player enters the server while the mods are loading?

----

> Geyser has an equal start message as minecraft so msh thinks the server is startet while other plugins are loading. 

bacause msh sets the server status as online in [these lines](https://github.com/gekigek99/minecraft-server-hibernation/blob/dev/lib/servctrl/cmd.go#L152-L159)

I could maybe  set the flag to `": Done"` instead of `"Done"` 

> While that happens players try to join and msh registers as joined.

because as of right now msh is counting players on the flags `"UUID of player"` and `"left the game"` as you can see [here](https://github.com/gekigek99/minecraft-server-hibernation/blob/dev/lib/servctrl/cmd.go#L199-L209)

**Comment by debian-user-france1** [comment]
correct, this is the line when server is opened for all. Maybe bedrock users can join bevore but I don't know. Geyser is a packet translator to let bdrck users join

**Comment by debian-user-france1** [comment]
> can you show a log example of when a player enters the server while the mods are loading?
> 
> > Geyser has an equal start message as minecraft so msh thinks the server is startet while other plugins are loading.
> 
> bacause msh sets the server status as online in [these lines](https://github.com/gekigek99/minecraft-server-hibernation/blob/dev/lib/servctrl/cmd.go#L152-L159)
> 
> I could maybe set the flag to `": Done"` instead of `"Done"`
> 
> > While that happens players try to join and msh registers as joined.
> 
> because as of right now msh is counting players on the flags `"UUID of player"` and `"left the game"` as you can see [here](https://github.com/gekigek99/minecraft-server-hibernation/blob/dev/lib/servctrl/cmd.go#L199-L209)

Could you please change that to the final message when a player really is in? Maybe a player just leaves and then no further information as I know or he gets kicked as my bug report with ?msh thinks server is .......

**Comment by gekigek99** [comment]
> change that to the final message when a player really is in?

what final message?

> Maybe a player just leaves and then no further information as I know or he gets kicked as my bug report with ?msh thinks server is .......

sorry, could you explain a little? i don't think I get what you mean...

----

just to be clear and speed things up:  
msh should set the server status to "online" only when it sees the line `Done (74.035s)! For help, type "help"` _(from the above example)_ so that people are able to join only after that...  
is this what you mean?

**Comment by debian-user-france1** [comment]
https://cloud.bubicloud.ch/nextcloud/index.php/s/LBMJLA6AL2pdLk2 <- log
and...
Once I saw msh was executing /players wich is 1. very wrong you have to do without / and 2. does not exist

**Comment by gekigek99** [comment]
> Once I saw msh was executing /players wich is 1. very wrong you have to do without / and 2. does not exist

sorry I don't get what you mean...

I fixed the code where msh executes commands on the server using the `/` as prefix (in dev branch)

**Comment by gekigek99** [comment]
Can you specify at what line is the server online (meaning players can join)?  
_(I see that there are 4 different `done` in the log and cannot understand which is from the plugin and which is from the server)_

**Comment by gekigek99** [comment]
the bugfix was implemented in f76e1d9ac12ff9317c1e2d3dccc2dbfd97484d16

please confirm that it is working as expected

**Comment by debian-user-france1** [comment]
I close this. Everything please continue there
